framework/src/systems/NonlinearSystemBase.C

```diff
@@ -1130,9 +1137,18 @@ NonlinearSystemBase::setConstraintSecondaryValues(NumericVector<Number> & soluti
             // reinit variables on the primary element's face at the contact point
             _fe_problem.setNeighborSubdomainID(primary_elem, 0);
             subproblem.reinitNeighborPhys(primary_elem, primary_side, points, 0);
+            if (displaced)
+              _fe_problem.reinitNeighborPhys(
+                  _mesh.elemPtr(primary_elem->id()), primary_side, points, 0);
+            else
+              subproblem.reinitNeighborPhys(primary_elem, primary_side, points, 0);
```

Comment:
Wow there's so much `reinitNeighborPhys` going on here 😅 In the undisplaced case, the exact same routine is going to be called twice in a row. Seems like lines 1143 and 1144 should be deleted

Reply:
I was testing SAM test with this clumsy code. The test did go into the displaced condition (feproblem call) and failed.